aws/ec2metadata/api.go

```diff
@@ -63,6 +64,42 @@ func (c *EC2Metadata) GetInstanceIdentityDocument() (EC2InstanceIdentityDocument
 	return doc, nil
 }
 
+// IAMInfo retrieves IAM info from the metadata API
+func (c *EC2Metadata) IAMInfo() (EC2IAMInfo, error) {
+	resp, err := c.GetMetadata("iam/info")
+	if err != nil {
+		return EC2IAMInfo{},
+			awserr.New("EC2RoleRequestError",
```

Comment:
Just realized `IAMInfo` and `GetInstanceIdentityDocument` use `EC2RoleRequestError` as their error code.  We probably should standardize error code for metadata requests as , `EC2MetadataRequestError`